### PR TITLE
fix(taskframework): add alarm in taskframework

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
module-taskframework
#### What this PR does / why we need it:
add alarm in taskframework
1. `TASK_CANCELED_FAILED` represent job be canceled failed
2. `TASK_HEART_TIMEOUT` represent job heart timeout
3.  `TASK_START_FAILED` represent job start failed 
4. `TASK_EXECUTOR_DESTROY_FAILED` represent destroy pod failed

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/alarm/AlarmEventNames.java [hunk 1]
```diff
@@ -56,4 +56,11 @@ private AlarmEventNames() {}
 
     public static final String SCHEDULING_FAILED = "SCHEDULING_FAILED";
 
+    /**
+     * TASK
+     */
+    public static final String TASK_CANCELED_FAILED = "TASK_CANCELED_FAILED";
+    public static final String TASK_HEART_TIMEOUT = "TASK_HEART_TIMEOUT";
```
Comment (yizhouxw) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/alarm/AlarmEventNames.java:63]: HEART --> HEARTBEAT

Comment (krihy) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/alarm/AlarmEventNames.java]: okey, done

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by ungreat — DISMISSED** [review]
LGTM

File: server/odc-core/src/main/java/com/oceanbase/odc/core/alarm/AlarmEventNames.java [hunk 2]
```diff
@@ -56,4 +56,11 @@ private AlarmEventNames() {}
 
     public static final String SCHEDULING_FAILED = "SCHEDULING_FAILED";
 
+    /**
+     * TASK
+     */
+    public static final String TASK_CANCELED_FAILED = "TASK_CANCELED_FAILED";
```
Comment (guowl3) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/alarm/AlarmEventNames.java:62]: I think adding a task failure alarm would help developers understand the task success rate and facilitate code optimization.

**Review by guowl3 — APPROVED** [review]
LGTM

**Review by LuckyPickleZZ — APPROVED** [review]

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**